cpp/ps0004/ps0004_solution.cpp

```diff
@@ -2,39 +2,32 @@
 using namespace std;
 
 int main() 
-
 {
-    int input;
-
-
+    // prompting user for number 1/1.5
+    float input;
     cout << "Enter a number: ";
+    cin >> input;
 
-
-    if (!(cin >> input)) 
+    // checking invalidaty of user's input 2.5/4
+    if (input == 0) 
     {
         cout << "Invalid character/s or zero value!" << endl;
-            }
+    }
 
-    if (input >= 32 && input <= 126) 
+    else if (static_cast<int>(input) != input) 
     {
-        cout << "The character for " << input << " is " << static_cast<char>(input) << "." << endl;
-
+        cout << "Invalid: a floating point number!" << endl;
     }
-
+    
     else if (input < 32 || input > 126) 
-
     {
         cout << "Invalid: number is out of the range!" << endl;
     }
-
-    else if (static_cast<int>(input) != input) 
-    {
-        cout << "Invalid: a floating point number!" << endl;
-    }
-
-    else 
+    
+    else
     {
-        cout << "Invalid character/s or zero value!" << endl;
+        // printing the number and the opposite printable character 2/2
+        cout << "The character for " << input << " is " << static_cast<char>(input) << "." << endl;
     }

```

Comment:
- [ ] used `if` structure in a right way
- [x] `if` structure take in consideration all invalid cases
- [x] all expressions to check invalidity are correct
- [x] used `std::cout` and `std::endl` or `\n` to print a different message for the 3 cases
- [x] converted the number to the opposite printable character
- [x] printed the number and the opposite printable character